clang/lib/StaticAnalyzer/Core/CallEvent.cpp

```diff
@@ -440,7 +440,6 @@ static SVal castArgToParamTypeIfNeeded(const CallEvent &Call, unsigned ArgIdx,
     return ArgVal;
 
   const FunctionDecl *Definition = CallExprDecl;
-  Definition->hasBody(Definition);
```

Comment:
The removal of this would break the `clang/test/Analysis/region-store.c:79:3` test case. It's intentional.
The `hasBody` (awkwardly though), sets the parameter (`Definition`) to the definition decl of the function decl if it has definition.

I could have used the `if (const auto *Def = CallExprDecl.getDefinition()) Definition = Def;` but that would have been longer to spell, along with a controlflow statement.

Reply:
These tests can be run using `check-clang-analysis`, or using `check-clang`.

Reply:
Yes, that makes sense. 
This change is wrong, and that's why I quickly closed the PRs associated with this. 
Thank you for your review.